src/Microsoft.AspNet.Http.Extensions/ResponseHeaders.cs

```diff
@@ -126,20 +126,20 @@ public EntityTagHeaderValue ETag
             }
         }
 
-        public UriHelper Location
+        public Uri Location
         {
             get
             {
                 Uri uri;
                 if (Uri.TryCreate(Headers[HeaderNames.Location], UriKind.RelativeOrAbsolute, out uri))
                 {
-                    return new UriHelper(uri);
+                    return uri;
                 }
                 return null;
             }
             set
             {
-                Headers.Set(HeaderNames.Location, value);
+                Headers.Set(HeaderNames.Location, value == null ? null : UriHelper.Encode(value));
```

Comment:
That only works for the reverse case `value == null ? DoSomethingElse() : null;` We only want to call Encode if it is not null.